src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -222,15 +222,20 @@ public FireStoreDirectoryEntry retrieveByPath(
     }
 
     public List<String> validateRefIds(
-        Firestore firestore, String collectionId, Collection<String> refIdArray)
+        Firestore firestore, String collectionId, List<String> refIdArray)
         throws InterruptedException {
 
+        int batchSize = configurationService.getParameterValue(FIRESTORE_RETRIEVE_BATCH_SIZE);
+        List<List<String>> batches = ListUtils.partition(refIdArray, batchSize);
```

Comment:
Same question as the other PR: Is there any concern essentially doubling the memory use for a large list of refIds?